app/logic/users.py

```diff
@@ -10,28 +12,58 @@ def isEligibleForProgram(program, user):
     :param user: accepts a User object or userid
     :return: True if the user is not banned and meets the requirements, and False otherwise
     """
-
-    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program).exists()):
+    now = datetime.datetime.now()
+    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program, ProgramBan.endDate > now).exists()):
         return False
 
     return True
 
-def addRemoveInterest(rule, program_id, currentUser):
+def addRemoveInterest(rule, program_id, username):
     """
     This function is used to add or remove interest from the interest table.
     Parameters:
     rule: Gets the url from the ajax call, specifies to add or remove interest
     program_id: id of the program the user is interested in
     """
-
     if 'addInterest' in str(rule):
-        Interest.get_or_create(program = program_id, user = currentUser)
+        Interest.get_or_create(program = program_id, user = username)
         return "Successfully added interest"
 
     elif 'deleteInterest' in str(rule):
         try:
-            deleted_interest = Interest.get(Interest.program == program_id, Interest.user == currentUser)
+            deleted_interest = Interest.get(Interest.program == program_id, Interest.user == username)
             deleted_interest.delete_instance()
             return "Successfully removed interest"
         except:
             return "This interest does not exist"
+
+
+def banUnbanUser(banOrUnban, program_id, username, note, banEndDate, creator):
```

Comment:
I would create a banUser method and an unbanUser method. And don't forget your tests!

Reply:
Different methods have been implemented.